geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -46,19 +52,68 @@
 public class RedisHash extends AbstractRedisData {
   public static final RedisHash NULL_REDIS_HASH = new NullRedisHash();
   private HashMap<ByteArrayWrapper, ByteArrayWrapper> hash;
+  private ConcurrentHashMap<UUID, List<ByteArrayWrapper>> hScanSnapShots;
+  private ConcurrentHashMap<UUID, Long> hScanSnapShotCreationTimes;
+  private ScheduledExecutorService HSCANSnapshotExpirationExecutor = null;
+  private final int HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS;
+  private final int MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE;
+
+  // for serialization
+  public RedisHash() {
+    this.hash = new HashMap<>();
+    this.hScanSnapShots = new ConcurrentHashMap<>();
+    this.hScanSnapShotCreationTimes = new ConcurrentHashMap<>();
+
+    this.HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS =
+        Integer.getInteger("redis.hscan-snapshot-cleanup-interval", 30000);
+
+    this.MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE =
+        Integer.getInteger("redis.hscan-snapshot-expiry", 30000);
```

Comment:
System properties like this should be set as `static final` instead of in the constructor since they are only intended to be set once for the life of the JVM. Also related is a comment in the associated test class.

Reply:
ok- reverted